cpp/src/io/parquet/page_string_decode.cu

```diff
@@ -625,8 +625,12 @@ __global__ void __launch_bounds__(preprocess_block_size) gpuComputePageStringSiz
       break;
```

Comment:
I think here too we can short circuit the call to `totalDictEntriesSize` with the same computation as below.